src/group_cmd.cpp

```diff
@@ -278,14 +278,16 @@ const Livery *GetParentLivery(const Group *g)
  * Propagate a livery change to a group's children.
  * @param g Group.
  */
-void PropagateChildLivery(const Group *g)
-{
-	/* Company colour data is indirectly cached. */
-	for (Vehicle *v : Vehicle::Iterate()) {
-		if (v->group_id == g->index && (!v->IsGroundVehicle() || v->IsFrontEngine())) {
-			for (Vehicle *u = v; u != nullptr; u = u->Next()) {
-				u->colourmap = PAL_NONE;
-				u->InvalidateNewGRFCache();
+void PropagateChildLivery(const Group *g, bool reset_cache = true)
```

Comment:
Shouldn't the call at line 297/299 of this file pass `reset_cache`?

Reply:
Good point. I think I'll make it non-optional to be safe :)